Mathlib/CategoryTheory/Sites/Coherent/RegularSheaves.lean

```diff
@@ -27,65 +29,219 @@ namespace CategoryTheory
+  apply equalizer.hom_ext
+  aesop
+
+/-- An alternative phrasing of the explicit equalizer condition, using more categorical language. -/
+theorem equalizerCondition_iff_isIso_lift (P : Cᵒᵖ ⥤ Type*) : EqualizerCondition P ↔
+    ∀ (X B : C) (π : X ⟶ B) [EffectiveEpi π] [HasPullback π π], IsIso
+    (equalizer.lift (P.map π.op) (equalizerCondition_iff_isIso_lift_w P π)) := by
+  constructor
+  · intro hP X B π _ _
+    have h := hP.bijective_mapToEqualizer_pullback _ X B π
+    rw [← isIso_iff_bijective, mapToEqualizer_eq_comp] at h
+    exact IsIso.of_isIso_comp_right (equalizer.lift (P.map π.op)
+      (equalizerCondition_iff_isIso_lift_w P π))
+      (Types.equalizerIso _ _).hom
+  · intro hP
+    apply EqualizerCondition.mk
+    intro X B π _ _
+    rw [mapToEqualizer_eq_comp, ← isIso_iff_bijective]
+    infer_instance
+
+/-- `P` satisfies the equalizer condition iff its precomposition by an equivalence does. -/
+theorem equalizerCondition_iff_of_equivalence (P : Cᵒᵖ ⥤ D)
+    (e : C ≌ E) : EqualizerCondition P ↔ EqualizerCondition (e.op.inverse ⋙ P) :=
+  ⟨fun h ↦ equalizerCondition_precomp_of_preservesPullback P e.inverse h, fun h ↦
+    equalizerCondition_of_natIso (e.op.funInvIdAssoc P)
+      (equalizerCondition_precomp_of_preservesPullback (e.op.inverse ⋙ P) e.functor h)⟩
+
+open WalkingParallelPair WalkingParallelPairHom in
+theorem parallelPair_pullback_initial {X B : C} (π : X ⟶ B)
+    (c : PullbackCone π π) (hc : IsLimit c) :
+    (parallelPair (C := (Sieve.ofArrows (fun (_ : Unit) => X) (fun _ => π)).arrows.categoryᵒᵖ)
+    (Y := op ((Presieve.categoryMk _ (c.fst ≫ π) ⟨_, c.fst, π, ofArrows.mk (), rfl⟩)))
+    (X := op ((Presieve.categoryMk _ π (Sieve.ofArrows_mk _ _ Unit.unit))))
+    (Quiver.Hom.op (Over.homMk c.fst))
+    (Quiver.Hom.op (Over.homMk c.snd c.condition.symm))).Initial := by
+  apply Limits.parallelPair_initial_mk
+  · intro ⟨Z⟩
+    obtain ⟨_, f, g, ⟨⟩, hh⟩ := Z.property
+    let X' : (Presieve.ofArrows (fun () ↦ X) (fun () ↦ π)).category :=
+      Presieve.categoryMk _ π (ofArrows.mk ())
+    let f' : Z.obj.left ⟶ X'.obj.left := f
+    exact ⟨(Over.homMk f').op⟩
+  · intro ⟨Z⟩ ⟨i⟩ ⟨j⟩
+    let ij := PullbackCone.IsLimit.lift hc i.left j.left (by erw [i.w, j.w]; rfl)
+    refine ⟨Quiver.Hom.op (Over.homMk ij (by simpa [ij] using i.w)), ?_, ?_⟩
+    all_goals congr
+    all_goals exact Comma.hom_ext _ _ (by erw [Over.comp_left]; simp [ij]) rfl
+
+/--
+Given a limiting pullback cone, the fork in `SingleEqualizerCondition` is limiting iff the diagram
+in `Presheaf.isSheaf_iff_isLimit_coverage` is limiting.
+-/
+noncomputable def isLimit_forkOfι_equiv (P : Cᵒᵖ ⥤ D) {X B : C} (π : X ⟶ B)
+    (c : PullbackCone π π) (hc : IsLimit c) :
+    IsLimit (Fork.ofι (P.map π.op) (equalizerCondition_w P c)) ≃
+    IsLimit (P.mapCone (Sieve.ofArrows (fun (_ : Unit) ↦ X) fun _ ↦ π).arrows.cocone.op) := by
+  let S := (Sieve.ofArrows (fun (_ : Unit) => X) (fun _ => π)).arrows
+  let E := @FullSubcategory (Over B) (fun f ↦ S f.hom)
```

Comment:
I believe that this proof can be cleaned up a bit due to the new API.

Reply:
It uses the `Presieve.category` API now. Did you have anything else in mind?